### PR TITLE
Add labels to preprocessing in TensorFlowV2Classifier.loss_gradient

### LINKED ISSUE #1004: Result of a targeted EoT attack with image rotation differs from Tutorial
Hi everybody,

I was trying to replicate the tutorial by [Athalye](https://www.anishathalye.com/2017/07/25/synthesizing-adversarial-examples/) (implemented in Tensorflow 1) by using the ART framework. My implementation is able to generate a misclassifcation, however, the result of the attack differs significantly from my implementation. The result from the tutorial looks like this:
cat_eot_rotation_tutorial

What I get is the following:
eot_attack_plot

Both results are generated with a rotation range of 45° and a sample size of 10. Besides, I use a pre-trained InceptionV3 neural network.

Moreover, I get different results after each execution of my code. I don't know how to fix this and maybe you guys can help me. Furthermore,  I didn't fully understand the difference between ```nb_samples``` in the ```EoTImageRotationTensorFlow``` class and ```sample_size``` in the ```ExpectationOverTransformations``` class

Here's the code I use:

```python
import json
import matplotlib.pyplot as plt
import numpy as np
import tensorflow as tf
import tensorflow_addons as tfa
from art.classifiers import TensorFlowV2Classifier
from art.wrappers.expectation import ExpectationOverTransformations
from art.preprocessing.expectation_over_transformation import EoTImageRotationTensorFlow
from art.attacks.evasion import ProjectedGradientDescent
from tensorflow.keras.applications.inception_v3 import InceptionV3


def plot_prediction(img, probs, correct_class=None, target_class=None):

    # Initialize the subplots
    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(10, 8))

    # Set the first plot to the input image
    fig.sca(ax1)
    ax1.imshow(img)

    # Determine the top ten labels and store them with their probabilities
    top_ten_indexes = list(probs[0].argsort()[-10:][::-1])
    print(top_ten_indexes)
    top_probs = probs[0, top_ten_indexes]
    print(top_probs)
    labels = [imagenet_labels[i] for i in top_ten_indexes]
    barlist = ax2.bar(range(10), top_probs)
    if target_class in top_ten_indexes:
        barlist[top_ten_indexes.index(target_class)].set_color('r')
    if correct_class in top_ten_indexes:
        barlist[top_ten_indexes.index(correct_class)].set_color('g')

    # Plot the probabilities and labels
    plt.sca(ax2)
    plt.ylim([0, 1.1])
    plt.xticks(range(10), labels, rotation='vertical')
    fig.subplots_adjust(bottom=0.2)
    plt.show()


# Step 1: Retrieve the labels of imagenet and load the image to be classified

with open('imagenet.json') as f:
    imagenet_labels = json.load(f)

image = tf.keras.preprocessing.image.load_img('cat.jpg')
image_class = np.array([281])
image = image.resize((299, 299)).crop((0, 0, 299, 299))
input_arr = tf.keras.preprocessing.image.img_to_array(image)
input_arr = np.array([input_arr])  # Convert single image to a batch.
input_arr = (input_arr / 255.0).astype(np.float32)

# Step 2: Load the pre-trained InceptionV3 model with ImageNet weights

model = InceptionV3(include_top=True, weights='imagenet', classifier_activation='softmax')

# Step 3: Create the TensorflowV2 Classifier

loss = tf.keras.losses.CategoricalCrossentropy(from_logits=False)
classifier = TensorFlowV2Classifier(model=model, nb_classes=1000, loss_object=loss,
                                    preprocessing=None, clip_values=(0.0,1.0), input_shape=(299,299,1))

# Step 4: Evalaute the Classifier on the adversarial image with a rotation

demo_epsilon = 8.0 / 255.0
demo_steps = 300
demo_target = np.array([924])  # Target class "guacamole". Has to be of shape (nb_samples,)

eot_rotation = EoTImageRotationTensorFlow(nb_samples=1, clip_values=(0.0,1.0), angles=22.5)

def trafo_rotation(x):
    rotation, _ = eot_rotation.forward(x=x, y=demo_target)
    return rotation


def transformation():
    yield trafo_rotation


ex_angle = tf.constant(np.pi/8)
rotated_tensor = tfa.image.rotate(input_arr, ex_angle)
rotated_image = np.squeeze(rotated_tensor)
eot = ExpectationOverTransformations(classifier=classifier, sample_size=10, transformation=transformation)
attack_eot = ProjectedGradientDescent(estimator=eot, eps=demo_epsilon * demo_steps, max_iter=demo_steps,
                                       eps_step=demo_epsilon, targeted=True)
adv_array = attack_eot.generate(x=input_arr, y=demo_target)
prediction = eot.predict(adv_array)
plot_prediction(rotated_image, prediction, correct_class=image_class, target_class=demo_target)
print('Done')
```
The json file is available [here](https://www.anishathalye.com/media/2017/07/25/imagenet.json) and the input image can be downloaded [here](https://www.anishathalye.com/media/2017/07/25/cat.jpg).

**System information (please complete the following information)**

- Windows 10
- Python 3.8.5
- ART 1.6.0
- Tensorflow 2.4.1
- tensorflow-addons 0.12.1

### PR DESCRIPTION
**Pull request by beat-buesser**
Signed-off-by: Beat Buesser <beat.buesser@ie.ibm.com>

# Description

This pull request fixes bug by adding labels to preprocessing in `TensorFlowV2Classifier.loss_gradient`.

Fixes #1004 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by minhitbk — APPROVED** [review]
ok
